selfservice/strategy/code/.schema/recovery.schema.json

```diff
@@ -11,8 +11,7 @@
       ]
     },
     "code": {
-      "type": "string",
-      "pattern": "^\\d{8}$"
```

Comment:
This was originally a suggestion by @zepatrik, but IMO proved impractical, as the error message from this is less than helpful ("does not match pattern ^\\d{8}$") and in the end, we "validate" this field by checking if the code is correct, and if someone enters a code that is not 8 (after this PR 6) characters long, then that code is invalid anyway.